test/e2e/framework/framework.go

```diff
@@ -156,3 +170,17 @@ func (f *Framework) CleanUp(t *testing.T, cleanupFunc func()) {
 		}
 	})
 }
+
+func (f *Framework) IsOpenshiftCluster() bool {
+	c, err := f.getOpenshiftClient()
+	if err != nil {
+		fmt.Printf("failed to get openshiftClient: %s", err)
+		return false
+	}
+	_, err = c.ConfigV1().ClusterVersions().Get(context.TODO(), "version", metav1.GetOptions{})
+	if err != nil {
```

Comment:
Not sure about this one. To me this reads we implicitly assume the error is a 404, in which case we can skip a test. If this returns a different error (say a 5xx) we would want to propagate an error. Otherwise we risk skipping the openshift tests even if we are in an openshift cluster. @simonpasquier @machine424 wdyt?

We might be better off to only return false `if apierrors.IsNotFound(err)`.